src/test/java/com/diffmin/AppTest.java

```diff
@@ -216,6 +230,66 @@ private static void runTests(TestResources sources) throws Exception {
                             .getOrCreate(retrievedFirstType);
             String patchedProgram = SpoonUtil.displayModifiedModel(patchedCtModel);
             assertEquals(cu.prettyprint(), patchedProgram, "Prev file was not patched correctly");
+
+            // check the root origination of each element
+            Set<String> insertedLines = new HashSet<>(Files.readAllLines(sources.testMetadata));
+
+            CtScanner scanner =
+                    new CtScanner() {
+                        private boolean doesElementBelongToGivenFile(
+                                CtElement element, String prefix) {
+                            return element.getPosition().getFile().getName().startsWith(prefix);
+                        }
+
+                        /** Checks if any of the members of a set collection are modified. */
+                        private boolean isSetCollectionModified(CtElement element) {
+                            if (element.getRoleInParent() == CtRole.THROWN) {
+                                return ((CtExecutable<?>) element.getParent())
+                                        .getThrownTypes().stream()
+                                                .anyMatch(
+                                                        thrownType ->
+                                                                doesElementBelongToGivenFile(
+                                                                        thrownType, NEW_PREFIX));
+                            }
+                            return false;
+                        }
+
+                        @Override
+                        public void scan(CtElement element) {
+                            if (element != null
+                                    && !element.isImplicit()
+                                    && element.getPosition().isValidPosition()) {
+                                String pathString = element.getPath().toString();
+                                // Check if element is a child of an element corresponding to new
+                                // path
```

Comment:
Inline comments saying _what_ the code does are for the most part bad and indicative that the code needs to be refactored. Inline comments saying _why_ the code does what it does are sometimes helpful if it's not clear by the context.

Most often, an inline comment saying what the code does can be removed if one extracts the thing that's commented into a separate method, and giving it a good name.

Reply:
To be clear, most (perhaps all) of the inline comments in this method are saying _what_ the code does.

Reply:
> Inline comments saying what the code does are for the most part bad and indicative that the code needs to be refactored.

Sounds good. I will try to refactor and rethink the names of the methods so that I don't need these.